cmake/module/GenerateHeaders.cmake

```diff
@@ -0,0 +1,21 @@
+# Copyright (c) 2023-present The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://opensource.org/license/mit/.
+
+function(generate_header_from_json json_source_relpath)
+  add_custom_command(
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${json_source_relpath}.h
+    COMMAND ${CMAKE_COMMAND} -DJSON_SOURCE_PATH=${CMAKE_CURRENT_SOURCE_DIR}/${json_source_relpath} -DHEADER_PATH=${CMAKE_CURRENT_BINARY_DIR}/${json_source_relpath}.h -P ${CMAKE_SOURCE_DIR}/cmake/script/GenerateHeaderFromJson.cmake
+    DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/${json_source_relpath} ${CMAKE_SOURCE_DIR}/cmake/script/GenerateHeaderFromJson.cmake
```

Comment:
When is `CMAKE_SOURCE_DIR`, `PROJECT_SOURCE_DIR` and `CMAKE_CURRENT_SOURCE_DIR` used here?
I'm not an expert in this area, but browsing a bit reveals that others were [often bitten](https://lesleylai.info/en/cmake_src_directory) by the differences among these.
Ignore, if the question doesn't make sense in this case...

Reply:
Addressed in https://github.com/hebasto/bitcoin/pull/301. Your review is welcome :)

Reply:
Thanks, ACK f9b48505d6beb2c4e388e275d89bad77a14de2bb

Reply:
Addressed in https://github.com/bitcoin/bitcoin/pull/30454#issuecomment-2271981040.